### PR TITLE
[C-796] Refactor AudiusAPIClient to work in native context

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Breaks apart audius-backend, audius-api-client, and the main store into common implementations. This prepares us to start integrating the first sagas into mobile!

### CODE REVIEW

File: packages/mobile/src/store/backend/sagas.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { call, put } from 'typed-redux-saga'
+
+import { audiusBackendInstance } from 'app/services/audius-backend-instance'
```
Comment (nicoback2) [inline packages/mobile/src/store/backend/sagas.ts:3]: i thought this would only work for web?

**Review by nicoback2 — APPROVED** [review]
LGTM besides one q

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-796-add-common-store-to-native

File: packages/mobile/src/store/backend/sagas.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/backend/sagas.ts:3]: so this audius-backend-instance is mobile compatible! but you're right that we shouldnt copy backend/sagas.ts here, and instead use the one in common (coming in a pr shortly)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-796-add-common-store-to-native

**Pull request merged**